### PR TITLE
"Preview with another theme" button

### PR DESCRIPTION
**Pull request by pazoozoo42**
https://user-images.githubusercontent.com/57953386/209395829-dc6757fc-665a-41f7-8a7b-bd0c13cf11e7.mp4

### CODE REVIEW

File: frontend/src/Components/ThemeToggleComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export default function ThemeToggleComponent(props: ThemeToggleComponentProps) {
+  const {theme, setTheme} = useTheme();
+  let initialTheme: string;
+
+  useEffect(() => {
+    if (theme) {
+      initialTheme = theme;
+      return () => {
```
Comment (Aivean) [inline frontend/src/Components/ThemeToggleComponent.tsx:21]: ☝️ what she said.

If the intention is to remember the original value of the `theme`, just create another state variable:

```
const [initialTheme] = useState(theme);
```

**Review by Aivean — CHANGES_REQUESTED** [review]
Looks good overall, please address the incorrect variable usage in effect.

**Comment by pazoozoo42** [comment]
> Looks good overall, please address the incorrect variable usage in effect.

@Aivean  done

File: frontend/src/Components/ThemeToggleComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  useEffect(() => {
+    if (theme) {
+      return () => {
+        if (props.resetOnOnmount) {
+          setTheme(initialTheme || 'light');
+        }
+      };
+    }
+  }, []);
```
Comment (Aivean) [inline frontend/src/Components/ThemeToggleComponent.tsx:26]:
```suggestion
    if (initialTheme) {
      return () => {
        if (props.resetOnOnmount) {
          setTheme(initialTheme || 'light');
        }
      };
    }
  }, [initialTheme, props.resetOnOnmount, setTheme]);
```

**Review by Aivean — APPROVED** [review]

**Pull request merged**